<a href="https://colab.research.google.com/github/dohaalnabahin/Data_science_and_machine_learning_Journey/blob/main/Intro_to_Time_Series_in_Python.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import datetime as dt
now = dt.datetime.now()

In [ ]:
## You can obtain the current date and time with .now()
now = dt.datetime.now()
now

datetime.datetime(2026, 6, 7, 21, 59, 14, 463941)

In [ ]:
## use datetime attribute to obtain just the year
now.year

2026

#**Datetime Formatting**

In [ ]:
## display using display function
display(now)

datetime.datetime(2026, 6, 7, 21, 59, 14, 463941)

In [ ]:
# display by leaving as final line in cell
now

datetime.datetime(2026, 6, 7, 21, 59, 14, 463941)

A `datetime` object like `datetime.datetime(2026, 6, 7, 21, 59, 14, 463941)` represents a specific point in time and contains several components:

*   **2026**: This is the **year**.
*   **6**: This is the **month** (June).
*   **7**: This is the **day** of the month.
*   **21**: This is the **hour** (21:00 is 9 PM).
*   **59**: This is the **minute**.
*   **14**: This is the **second**.
*   **463941**: This is the **microsecond** (one millionth of a second).

In [ ]:
# Assuming 'now' is your datetime object, you can access each part using its attributes:

print(f"Year: {now.year}")
print(f"Month: {now.month}")
print(f"Day: {now.day}")
print(f"Hour: {now.hour}")
print(f"Minute: {now.minute}")
print(f"Second: {now.second}")
print(f"Microsecond: {now.microsecond}")

Year: 2026
Month: 6
Day: 7
Hour: 21
Minute: 59
Second: 14
Microsecond: 463941


In [ ]:
print(now)

2026-06-07 21:59:14.463941


#**Datetime.strftime**

##For the day of the month:

In [ ]:
# just output the day with a leading 0 if needed
format_a = "%d"
now.strftime(format_a)

'07'

# For month:

In [ ]:
# Add full month name to day of month
format_b = "%B %d"
now.strftime(format_b)

'June 07'

# For the day of the week:

In [ ]:
# Add day of the week to month and day
format_c = "%a, %B %d"
now.strftime(format_c)

'Sun, June 07'

# For the year:

In [ ]:
# include abbreviated month and 2 digit year
format_d = "%b %y"
now.strftime(format_d)

'Jun 26'

# For Time:

In [ ]:
## the default datetime format
default ="%Y-%m-%d %H:%M:%S.%f"
print(now)
now.strftime(default)

2026-06-07 21:59:14.463941


'2026-06-07 21:59:14.463941'

In [ ]:
# Example US Formatted 12-hour date/time with AM/PM
fmt = "%m/%d/%Y %I:%M:%S %p"
now.strftime(fmt)

'06/07/2026 09:59:14 PM'

In [ ]:
## just the time (24 hour time)
now.strftime("%T")

'21:59:14'